### DISCUSSION TITLE
Supports play and play in non-debug mode

### DISCUSSION THREAD

**Discussion by CodeLazier** [godotengine/godot-proposals#5054] [Editor]
Fed up with the behavior of triggering a breakpoint after debugging and forgetting to remove the breakpoint, or being able to simply disable/enable all breakpoints. most ide and vscode can support both of these actions.

**Comment by Calinou** [comment]
This is already implemented in 3.5 and 4.0. You can skip all breakpoints while running the project by clicking the red circle button in the top-right corner of the Debugger bottom panel (highlighted with a blue rectangle around it on the screenshot):

![image](https://user-images.githubusercontent.com/180032/182482370-811373df-0d3c-4a49-8d0f-035f6f6eb79c.png)

**Reply by CodeLazier** [reply]
Yes but I don't feel intuitive enough. Because debugger panel defaults to a collapsed state, if I want to run non-debugging I have to click on debug tab, then click skip breakpoint, and then play project, the debug panel will collapse again by default after stop.

Because debug and non-debug runs are repeatedly switching operations in development. So I wanted to have at least one button to run enable skip breakpoint -> play project -> disable skip breakpoint  to speed up the debugging experience.

**Reply by Calinou** [reply]
If you don't intend to use any of the debugger features, you can [run the project from a terminal](https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html). This can be done while the editor is open.
